python/ray/data/dataset.py

```diff
@@ -2097,7 +2098,13 @@ def zip(self, other: "Dataset[U]") -> "Dataset[(T, U)]":
         """
 
         plan = self._plan.with_stage(ZipStage(other))
-        return Dataset(plan, self._epoch, self._lazy)
+
+        logical_plan = self._logical_plan
+        other_logical_plan = other._logical_plan
+        if logical_plan is not None and other_logical_plan is not None:
```

Comment:
We will either have both None, or both not None, right? We can check that if so.

Reply:
Actaully right now it's not the case. For the `Dataset` has API not supported as logical operator yet, the logical_plan will be None. So something like this:

```
ds1 = ray.data.from_items(...) <-- ds1._logical_plan is None
ds2 = ray.data.read_parquet(...) <-- ds2._logical_plan is not None
ds1.zip(ds2) <-- will fail if we add assertion here.
```

Reply:
I thought we added that. In that case it looks fine to require both non-none to create this Zip op.